docs/architecture.md

```diff
@@ -1,32 +1,139 @@
+
+> This hashing scheme was originally chosen to reduce the number of required ingesters on the query path. The trade-off, however, is that the write load on the ingesters is less even.
+
+#### The hash ring
+
+A consistent hash ring is stored in [Consul](https://www.consul.io/) as a single key-value pair, with the ring data structure also encoded as a [Protobuf](https://developers.google.com/protocol-buffers/) message. The consistent hash ring consists of a list of tokens and ingesters. Hash values are looked up in the ring; the replication set is built for the closest unique ingesters by token. One of the benefits of this system is that adding and remove ingesters results in only 1/_n_ of the series being moved (where _n_ is the number of ingesters).
+
+All distributors share access to the same hash ring, which means that write requests can be sent to any distributor.
+
+#### Quorum consistency
+
+To ensure consistent query results, Cortex uses DynamoDB-style quorum consistency on reads and writes. This means that the distributor will wait for a positive response of at least one half plus one of the ingesters to send the sample to before responding to the user.
+
+#### Load balancing across distributors
+
+We recommend randomly load balancing write requests across distributor instances, potentially by running the distributors as a Kubernetes [Service](https://kubernetes.io/docs/concepts/services-networking/service/).
+
+### Ingester
+
+The **ingester** service is responsible for writing sample data to long-term storage backends (DynamoDB, S3, Cassandra, etc.).
+
+#### Shared-nothing processes
+
+Ingesters are, for the most part, shared-nothing processes. They don't coordinate or communicate with each other *under normal operation*. The exception is on rolling updates, during which ingesters transfer their in-memory data from the ingester exiting the group to the ingester joining the group.
+
+#### State
+
+Ingesters are semi-stateful in that they always retain the last 12 hours worth of samples. When restarting or upgrading ingesters, care must be taken to avoid losing that data.
+
+As *semi*-stateful processes, ingesters are *not* designed to be long-term data stores. In Cortex, that role is played by the [chunk store](#chunk-store).
+
+#### Write de-amplification
+
+Ingesters semi-statefully store the last 12 hours worth of samples in order to perform **write de-amplification**, i.e. batching and compressing samples for the same series and flushing them out to the [chunk store](#chunk-store). Under normal operations, there should be *many* orders of magnitude fewer queries per second (QPS) worth of writes to the chunk store than to the ingesters.
+
+This division of labor between the ingester and the chunk store is the main source of Cortex's low total cost of ownership (TCO).
```

Comment:
Its not the division of labour but rather the write-deamplification - ie if you tried to write each sample to a Cassandra cluster, you'd need a cluster capable of millions of QPS.  With cortex, the same can be don't with a cassandra cluster capable of thousands of QPS.